### ISSUE TITLE
Evaluate: Benchmarks to compare with/without this proposal

### ISSUE THREAD

**Issue by alexcrichton** [WebAssembly/128-bit-arithmetic#3]
Original development of this proposal benchmarked [the `blind-sig` benchmark in Sightglass](https://github.com/bytecodealliance/sightglass/tree/main/benchmarks/blind-sig) as well as the fibonacci benchmarks from the Rust [num-bigint repository](https://github.com/rust-num/num-bigint). 

This issue is intended to serve as a location for others to drop interesting benchmark programs as well so they can be collected to help evaluate this proposal over time. If you've got a benchmark you'd like to see added it would ideally be in C or Rust at this time and is ideally a program that has a means of self-reporting its execution time. High-level ideas are ok to but would require some more work to create a reproducible benchmark.

**Comment by alexcrichton** [comment]
One area that would be particularly interesting to have benchmarks for are programs that require good performance of overflowing/saturating/checked arithmetic which isn't related to 128-bit. This would help stress the need for either 128-bit operations or overflow-flag-returning-instructions.

**Comment by alexcrichton** [comment]
A suggestion [here](https://github.com/WebAssembly/design/issues/1522#issuecomment-2251592505) is that `-ftrapv` can inject checked arithmetic for C and UBSan might rely on this heavily. A [naive benchmark](https://github.com/WebAssembly/design/issues/1522#issuecomment-2253539268) didn't show much performance difference relative to native without this proposal, however.

**Comment by marcusdarmstrong** [comment]
Just found out about this proposal today, but I've got a strong benchmark candidate here if you're still looking—[XXH3](https://github.com/Cyan4973/xxHash) is [critically reliant on a wide u64 mul](https://github.com/Cyan4973/xxHash/blob/dev/xxhash.h#L4354) operation for low-input-sizes, and thus our existing [manual WASM implementation](https://github.com/jungomi/xxhash-wasm) demonstrates [worse performance](https://github.com/jungomi/xxhash-wasm/issues/29) than the older, non-vectorized XXH64 algorithm.

**Comment by CryZe** [comment]
Yeah, it's not just XXH3, but rustc_hash, foldhash, wyhash, rapidhash, mumhash and many more that all rely on 128-bit widening multiplication.

**Comment by alexcrichton** [comment]
Thanks @marcusdarmstrong and @CryZe! It'll be a bit easier to test and confirm in a few months once rustc and LLVM both support wide-arithmetic, but `i64.mul_wide_u` should be perfect for 128-bit-widening-multiplication. Historical benchmarks have all shown that the wasm instructions are suitable for matching native performance in these situations.

**Comment by alexcrichton** [comment]
I'm going to close this now that the proposal is phase 4 and finalized
